lib/util/assignOptional.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information
+
```

Comment:
it'd be good to add a comment on why you need this function, here or in the place where you're calling it below